packages/@lwc/integration-karma/test/component/LightningElement.disconnectedCallback/index.spec.js

```diff
@@ -189,3 +191,17 @@ describe('disconnectedCallback for components with a explicit render()', () => {
         });
     });
 });
+
+describe('context', () => {
+    it('removing child unsubscribes from context subscription during disconnect', async () => {
+        const elm = createElement('x-context-parent', { is: ContextParent });
+        const state = nameStateFactory();
+        elm.state = state;
+        document.body.appendChild(elm);
+
+        expect(state.subscribers.size).toBe(1);
+        elm.hideChild = true;
+        await new Promise((resolve) => requestAnimationFrame(resolve));
+        expect(state.subscribers.size).toBe(0);
+    });
+});
```

Comment:
Personally I'd prefer to put all the context-related tests together in one folder, but it's not a big deal.